subgraphs/saddle-finance/src/entities/protocol.ts

```diff
@@ -0,0 +1,229 @@
+import {
+  Address,
+  BigDecimal,
+  dataSource,
+  ethereum,
+} from "@graphprotocol/graph-ts";
+import {
+  FinancialsDailySnapshot,
+  UsageMetricsDailySnapshot,
+  Account,
+  DexAmmProtocol,
+  UsageMetricsHourlySnapshot,
+  ActiveAccount,
+} from "../../generated/schema";
+import {
+  ProtocolType,
+  PROTOCOL_METHODOLOGY_VERSION,
+  PROTOCOL_NAME,
+  PROTOCOL_SCHEMA_VERSION,
+  PROTOCOL_SLUG,
+  PROTOCOL_SUBGRAPH_VERSION,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  ZERO_ADDRESS,
+} from "../utils/constants";
+
+export function getOrCreateProtocol(): DexAmmProtocol {
+  let protocol = DexAmmProtocol.load(ZERO_ADDRESS);
+  if (!protocol) {
+    // Zero address because no registry
+    protocol = new DexAmmProtocol(ZERO_ADDRESS);
```

Comment:
We should use a more meaningful address. Maybe address of the first deployer. Main reason is that at one point we may have a "mega" subgraph that'll contain all DEX-AMMs, and there needs to be a way to differentiate their IDs